app/scaffold-controllers/cartController.js

```diff
@@ -1,27 +1,63 @@
 define([
     'bluebird',
-    'plugins/webViewPlugin'
+    'config/baseConfig',
+    'config/cartConfig',
+    'plugins/webViewPlugin',
+    'plugins/anchoredLayoutPlugin',
+    'plugins/loaders/defaultLoaderPlugin',
+    'scaffold-controllers/cartHeaderController'
 ],
 /* eslint-disable */
 function(
     Promise,
-    WebViewPlugin
+    BaseConfig,
+    CartConfig,
+    WebViewPlugin,
+    AnchoredLayoutPlugin,
+    LoaderPlugin,
+    CartHeaderController
 ) {
 /* eslint-enable */
 
-    var CartController = function(webView) {
+    var CartController = function(headerController, layout, webView) {
+        this.viewPlugin = layout;
+
         this.webView = webView;
+        this.headerController = headerController;
     };
 
     CartController.init = function() {
         var webViewPromise = WebViewPlugin.init();
+        return Promise.join(
+            CartHeaderController.init(),
+            AnchoredLayoutPlugin.init(),
+            WebViewPlugin.init(),
+        function(headerController, layout, webView) {
+            var loader = webView.getLoader();
+            loader.setColor(BaseConfig.colors.primaryColor);
+            webView.navigate(CartConfig.url);
 
-        return webViewPromise.then(function(webView) {
-            webView.navigate('https://webpush-you-host.mobifydemo.io/cart/');
+            layout.addTopView(headerController.viewPlugin);
+            layout.setContentView(webView);
 
-            return new CartController(webView);
+            return new CartController(headerController, layout, webView);
         });
     };
 
+    CartController.prototype.registerCloseEventHandler = function(callback) {
+        if (!callback) {
+            return;
+        }
+        this.headerController.registerCloseEventHandler(callback);
+    };
+
+    CartController.prototype.navigate = function(url) {
+        this.webView.navigate(url);
+    };
+
+    CartController.prototype.reload = function() {
+        this.webView.navigate(CartConfig.url);
```

Comment:
Right, makes sense. I just tried out the `navigationPlugin.reload()` and it seems to work for me (though we did mark it as a deprecated method since we want you calling `reload` on WebViewPlugins instead. Anyways, if it appears broken again let me know and we can look into the individual case together.